arangod/VocBase/Methods/Collections.cpp

```diff
@@ -573,15 +565,13 @@ Result Collections::drop(TRI_vocbase_t* vocbase, LogicalCollection* coll,
 Result Collections::warmup(TRI_vocbase_t& vocbase,
                            LogicalCollection const& coll) {
   ExecContext const* exec = ExecContext::CURRENT;  // disallow expensive ops
-
-  if (!exec->isSuperuser() && ServerState::readOnly()) {
+  if (!exec->canUseCollection(coll.name(), auth::Level::RW)) {
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_READ_ONLY,
                                    "server is in read-only mode");
```

Comment:
The error message may be wrong here.
The error can be thrown when the server is in read-only mode but also when the user does not have write privileges on the collection.
Can't we just return TRI_ERROR_FORBIDDEN?